src/SFA.DAS.Commitments.Infrastructure/Data/DataLockRepository.cs

```diff
@@ -88,16 +89,21 @@ private static bool IsConstraintError(SqlException ex)
             return ex.Errors?.Count == 2 && ex.Errors[0].Number == 547;
         }
 
-        public async Task<List<DataLockStatus>> GetDataLocks(long apprenticeshipId)
+        public async Task<List<DataLockStatus>> GetDataLocks(long apprenticeshipId, bool includeRemoved)
         {
             return await WithConnection(async connection =>
             {
+                var sql = new StringBuilder();
+                sql.Append($"SELECT * FROM DataLockStatus WHERE ApprenticeshipId = @ApprenticeshipId ");
+                if (!includeRemoved) sql.Append("AND EventStatus <> 3 AND IsExpired = 0 ");
+                sql.Append("ORDER BY IlrEffectiveFromDate, Id");
+
                 var parameters = new DynamicParameters();
                 parameters.Add("@ApprenticeshipId", apprenticeshipId);
                 var results = await connection.QueryAsync<DataLockStatus>(
-                    sql: $"[dbo].[GetDataLockStatusesByApprenticeshipId]",
```

Comment:
Hej,

Do we need to delete GetDataLockStatusesByApprenticeshipId SP in the next release if we dont use it anymore?

Maybe it would be better to pull back all the datalocks for the apprenticeship and then filter in the C# code? Should never be that much data?
/c

Reply:
Proc - a good point sir, I will remove it (you know I hate procs).

Not sure about the filter in C#, makes little difference?

Reply:
Maybe its more clear in C# (I think) but it not a big deal. 

Don't remove it yet. Guess its needed in the deployment.